### PR TITLE
SonarCloud guided improvements

### PR DESCRIPTION
**Pull request by dlmiles**
These are a little more involved changes, to better understand the context and decide on a strategy.
Most commits take a defensive coding approach to scenario SonarCloud may present.

### CODE REVIEW

File: graphics/grOGL1.c [hunk 1]
```diff
@@ -1076,7 +1078,8 @@ GrOGLIconUpdate(w,text)
 
      if (brack = strchr(text,'['))
      {
-     	  brack--;
+	  if (brack != text)
```
Comment (dlmiles) [inline graphics/grOGL1.c:1081]: Consider if this should move up into the main if() 2 lines above - note to self

**Comment by dlmiles** [comment]
Resolved my own query slightly differently.  Found a total of 5 sites with similar pattern, these updated too.

**Review by RTimothyEdwards — APPROVED** [review]
All of these changed look correct.

**Comment by RTimothyEdwards** [comment]
Pulled and merged into opencircuitdesign.com code base.

**Pull request closed without merging**